### ISSUE TITLE
OctoBot behind nginx reverse proxy

### ISSUE THREAD

**Issue by jon4hz** [Drakkar-Software/OctoBot#1308]
Did anyone manage to run octobot behind a reverse proxy? It seemed to work at first but then octobot lost the connection after a few seconds and couldn't reconnect.

**Comment by Herklos** [comment]
Hi @jon4hz, yes the websocket lost the connection if your nginx configuration doesn't take into account the websocket or doesn't properly redirect the related port.

**Comment by jon4hz** [comment]
Hey, thanks for the hint with the websocket. Sadly I couldn't get it to work (yet). My config looks like this:
```
server {
        listen 80;
        server_name <servername>;
        return 301 https://$server_name$request_uri;
}



server {
        listen 443 ssl;
        server_name <servername>;

        access_log /config/log/nginx/access.log;
        error_log /config/log/nginx/error.log;


        location / {

                                proxy_pass http://octobot:5001;

                                proxy_http_version 1.1;
                                proxy_set_header Upgrade $http_upgrade;
                                proxy_set_header Connection \"upgrade\";
                                proxy_set_header Host $host;
                                proxy_read_timeout 86400;
        }
}
```

Can anyone help me out? I'm not really a nginx expert.. Thank a lot ^^

**Comment by Herklos** [comment]
I'm not an expert either but you should probably have a look at this stackoverflow : https://stackoverflow.com/questions/49802143/flask-socketio-hidden-behind-reverse-proxy-with-nginx :wink:

**Comment by jon4hz** [comment]
Still doesn't work :(

It seems like the http push method does not work (Error 400)

**Comment by Herklos** [comment]
Hi @jon4hz, can you add `proxy_set_header Connection "upgrade";` and retest please ?

**Comment by jon4hz** [comment]
I never applied those "\". I guess they were added by pasting the code to github... sorry for the confusion

**Comment by Herklos** [comment]
Socketio usage is now fixed behind a proxy.

**Comment by amasu** [comment]
> Socketio usage is now fixed behind a proxy.

Hi, I was having the same issue with nginx + ssl until 0.4.0-stable. I confirm the fix on 0.4.0-beta2.

Thank you :)

**Comment by Herklos** [comment]
> > Socketio usage is now fixed behind a proxy.
> 
> Hi, I was having the same issue with nginx + ssl until 0.4.0-stable. I confirm the fix on 0.4.0-beta2.
> 
> Thank you :)

Thanks @amasu !
